querybook/webapp/components/QueryPeerReviewModal/QueryPeerReviewModal.tsx

```diff
@@ -0,0 +1,207 @@
+import { AsyncButton } from 'ui/AsyncButton/AsyncButton';
+import { FormField } from 'ui/Form/FormField';
+import { FormWrapper } from 'ui/Form/FormWrapper';
+import { SimpleField } from 'ui/FormikField/SimpleField';
+import { Link } from 'ui/Link/Link';
+import { Message } from 'ui/Message/Message';
+import { Modal } from 'ui/Modal/Modal';
+import { IStandardModalProps } from 'ui/Modal/types';
+
+import './QueryPeerReviewModal.scss';
+
+interface IQueryPeerReviewFormProps {
+    query: string;
+    queryEngine: IQueryEngine;
+    onConfirm: (
+        reviewerIds: number[],
+        externalRecipients: string[],
+        notifierName: string,
+        justification: string,
+        query: string,
+        queryEngine: IQueryEngine
+    ) => Promise<void>;
+    onHide: () => void;
+}
+
+const QueryPeerReviewForm: React.FC<IQueryPeerReviewFormProps> = ({
+    query,
+    queryEngine,
+    onConfirm,
+    onHide,
+}) => {
+    const notifiers = useSelector(notificationServiceSelector);
+    const notifierOptions = useMemo(
+        () =>
+            notifiers.map((notifier) => ({
+                value: notifier.name,
+                label: notifier.name,
+            })),
+        [notifiers]
+    );
+    const getNotifierHelp = useCallback(
+        (notifierName: string) =>
+            notifiers.find((n) => n.name === notifierName)?.help ||
+            'Add comma(,) separated recipients here',
+        [notifiers]
+    );
+
+    const initialValues = {
+        notifyWith: '',
+        reviewers: [],
+        justification: '',
+    };
+
+    const peerReviewFormSchema = Yup.object().shape({
+        notifyWith: Yup.string().required(
+            'Please select a notification method'
+        ),
+        reviewers: Yup.array()
+            .min(1, 'Please select at least one reviewer')
+            .required('Please select at least one reviewer'),
+        justification: Yup.string().required('Justification is required'),
+    });
+
+    const handleSubmit = useCallback(
+        async (values) => {
+            try {
+                const notifierName = values.notifyWith;
+                const reviewerIds = values.reviewers
+                    .filter((v) => 'isUser' in v && v.isUser)
+                    .map((v) => v.value);
+
+                const externalRecipients = values.reviewers
+                    .filter((v) => !('isUser' in v) || !v.isUser)
+                    .map((v) => v.value);
+
+                await onConfirm(
+                    reviewerIds,
+                    externalRecipients,
+                    notifierName,
+                    values.justification,
+                    query,
+                    queryEngine
+                );
+                onHide();
+                toast.success(
+                    'Review request sent! Your reviewers have been notified, and if approved, your query will be executed.'
+                );
```

Comment:
after sending the request, where can the user to check all the requested executions and their status?

Reply:
Good point, I will need to create 2nd tab in Request Review Modal above to monitor existing all reviews. 
I think we can also add a link in the QueryExecutionBar to review. Will handle these in later PR